assignments/002-unix-intro/unix-intro.sh

```diff
@@ -0,0 +1,708 @@
+
+
+#### Ex. 48: Print the _first_ 15 lines of `sam/DPCh_plate1_A05_S5.sam` to _stdout_
+
+head -n 15 sam/DPCh_plate1_A05_S5.sam
+
+-----
+
+@HD	VN:1.3	SO:coordinate
+@SQ	SN:NC_037097.1	LN:96198142
+@SQ	SN:NC_037098.1	LN:57406636
+@SQ	SN:NC_037099.1	LN:81079776
+@SQ	SN:NC_037100.1	LN:74299616
+@SQ	SN:NC_037101.1	LN:93271540
+@SQ	SN:NC_037102.1	LN:77009353
+@SQ	SN:NC_037103.1	LN:86953105
+@SQ	SN:NC_037104.1	LN:73297397
+@SQ	SN:NC_037105.1	LN:90872356
+@SQ	SN:NC_037106.1	LN:60898746
+@SQ	SN:NC_037107.1	LN:48052637
+@SQ	SN:NC_037108.1	LN:77127610
+@SQ	SN:NC_037109.1	LN:75288599
+@SQ	SN:NC_037110.1	LN:55471129
+
+#### Ex. 49: Catenate the file `count_files.sh` to _stdout_
+
+cat count_files.sh
+
+-----
+
+# This looks fun
+ls -lR  * | awk 'NF>7' | wc | awk '{print "Estimated", $1, "files in the current directory"}'
+
+
+#### Ex. 50: Note that `count_files.sh` is a short shell script. Try to 
+####         execute it with `./count_files.sh`
+
+./count_files.sh
+
+-----
+
+Estimated 77 files in the current directory
+
+
+
+#### Ex. 51: List (long) `count_files.sh` to see what the permissions are,
+####         and then add execute (`x`) permissions to users and groups with
+####         the "add-or-subtract permissions syntax" of `chmod`, then long list it again
+
+
+ls -l count_files.sh
+
+-----
+
+-rwxr-xr-x. 1 ldurkee@colostate.edu ldurkeegrp@colostate.edu 112 Jan 23 10:43 count_files.sh
+
+
+chmod +x count_files.sh
+
+ls -l count_files.sh
+
+-----
+
+-rwxr-xr-x. 1 ldurkee@colostate.edu ldurkeegrp@colostate.edu 112 Jan 23 10:43 count_files.sh
+
+#### Ex. 52: Use chmod to change permissions of the files in the `silly`
+####         directory to match their names. For example for the first 
+####         one you would use: `chmod 770 FILE-rwxrwx---`. Use the octal
+####         specifiers to `chmod`. Do this for all 5 files (one chmod command for each),
+####         and at the end do `ls -l silly` to verify they are correct.
+
+chmod 711 silly/DIRdrwxrwxr-x
+chmod 770 silly/FILE-rw-r-----
+chmod 664 silly/FILE-rw-rw-r--
+chmod 711 silly/FILE-rwxr-----
+chmod 764 silly/FILE-rwxrwxr--
+
+
+ls -l silly
+
+-----
+
+total 128
+drwx--s--x. 2 ldurkee@colostate.edu ldurkeegrp@colostate.edu 23 Jan 23 10:43 DIRdrwxrwxr-x
+-rwxrwx---. 1 ldurkee@colostate.edu ldurkeegrp@colostate.edu  0 Jan 23 10:43 FILE-rw-r-----
+-rw-rw-r--. 1 ldurkee@colostate.edu ldurkeegrp@colostate.edu  0 Jan 23 10:43 FILE-rw-rw-r--
+-rwx--x--x. 1 ldurkee@colostate.edu ldurkeegrp@colostate.edu  0 Jan 23 10:43 FILE-rwxr-----
+-rwxrw-r--. 1 ldurkee@colostate.edu ldurkeegrp@colostate.edu  0 Jan 23 10:43 FILE-rwxrwxr--
+
+# I don't really understand this one, and I'm not sure if I did it correctly. I used ChatGPT to help!
```

Comment:
ChatGPT didn't quite get this one!  

chmod 775 silly/DIRdrwxrwxr-x/
chmod 640 silly/FILE-rw-r-----
chmod 664 silly/FILE-rw-rw-r--
chmod 740 silly/FILE-rwxr-----
chmod 774 silly/FILE-rwxrwxr--

The bits are:

1 = execute
2 = write
4 = read

So:
1 = execute
3 = execute and write,
4 = read
5 = read and execute
6 = read and write
7 = read write and execute

The three numbers are user, group, other

So 664 is user read-write, group read-write, and other read. etc.

We can discuss in class